csharp/native/HelloWorld/Client/Program.cs

```diff
@@ -0,0 +1,178 @@
+    /// <returns>
+    ///   Task representing the asynchronous execution of the method
+    /// </returns>
+    /// <exception cref="Exception">Issues with results from tasks</exception>
+    /// <exception cref="ArgumentOutOfRangeException">Unknown response type from control plane</exception>
+    internal static async Task Run(string endpoint,
+                                   string partition)
+    {
+      // Create gRPC channel to connect with ArmoniK control plane
+      var channel = GrpcChannelFactory.CreateChannel(new GrpcClient
+                                                     {
+                                                       Endpoint = endpoint,
+                                                     });
+
+      // Create client for task submission
+      var submitterClient = new Submitter.SubmitterClient(channel);
+
+      // Request for session creation with default task options and allowed partitions for the session
+      var createSessionReply = submitterClient.CreateSession(new CreateSessionRequest
+                                                             {
+                                                               DefaultTaskOption = new TaskOptions
+                                                                                   {
+                                                                                     MaxDuration = Duration.FromTimeSpan(TimeSpan.FromHours(1)),
+                                                                                     MaxRetries  = 2,
+                                                                                     Priority    = 1,
+                                                                                     PartitionId = partition,
+                                                                                   },
+                                                               PartitionIds =
+                                                               {
+                                                                 partition,
+                                                               },
+                                                             });
+
+      // Create the result id
+      var resultId = Guid.NewGuid()
+                         .ToString();
+
+      var createTaskReply = await submitterClient.CreateTasksAsync(createSessionReply.SessionId,
+                                                                   null,
+                                                                   new[]
+                                                                   {
+                                                                     // Task request with payload for the task
+                                                                     // Also contains the list of results that will be created by the task
+                                                                     new TaskRequest
```

Comment:
I would create a temporary object just to make it easier to read.

Reply:
Not really convinced on that one

Reply:
It would be more readable without the "weird" indentation. The temporary variable would mitigate the indentation issue.